### DISCUSSION TITLE
[Problem] Getting two hover menus, sometimes - Conflict with mason-lspconfig

### DISCUSSION THREAD

**Discussion by Ciel-MC** [mrcjkb/haskell-tools.nvim#304] [Q&A] [answered]
I'm trying learning haskell and trying to do my editor setup, but it seems like when I try to `K` something, the plugin's hover pops up, but sometimes there's a second one that sometimes shows the type? And if I press `K` again I go into that one instead of the one that has actions. I see that it's from HLS but I can't have that not launch because then things like formatting don't work, and for some reason if I let it start and then kill it, things seem to work fine, format is there and the extra hover menu isn't, I'm using mason + lspconfig + mason-lspconfig, please lmk if there's some known conflicts here, thanks

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

This looks like a conflict caused by mason-lspconfig,
which sets up the `lspconfig.hls` LSP client configuration. As a result, you have two LSP clients attached to the buffer.

You can confirm this with `:checkhealth haskell-tools`.

To solve this, you have to either disable the `lspconfig.hls` configuration or configure mason-lspconfig to not set it up.

**Reply by Ciel-MC** [reply]
That's what I guessed as well, and I tried just returning early like this, but it seems also make hs-tools not be able to start up the server :sweat_smile:, do you mind telling me what the right way to do it is?
![image](https://github.com/mrcjkb/haskell-tools.nvim/assets/47144701/86c3de2e-7428-4d64-beb6-bd3ab7629e54)

**Reply by Ciel-MC** [reply]
Alright I figured it out, since ht-tools is using the system install, there was no point in having hls installed with Mason in the first place, uninstalling that fixed the issue(though I would like to see a possible integration to use mason's version so a local install is not required).

**Reply by mrcjkb** [reply]
If you add `haskell-language-server` to mason.nvim's  `ensure_installed` option, it should pick it up automatically.
At least it seems to work in a [LazyVim PR](https://github.com/LazyVim/LazyVim/pull/2052/files) someone requested.

I don't support mason explicitly, but if it doesn't pick up hls, you can specify the command with the `vim.g.haskell_tools.hls.cmd` option (see `:h haskell-tools.config`).
